File: matlab_package/brainflow/examples/CSP.m
BoardShim.set_log_file('brainflow.log');
BoardShim.enable_dev_board_logger();

n_epochs = 2;
n_channels = 2;
n_times = 4;

% In this toy example mathematically we have two matrices (epochs) with rows as times and columns as channels
% Here data is a [n_channels, n_times, n_epochs] array
data = [6 3 1 5; 3 0 5 1];
data(:,:,2) = [1 5 6 2; 5 1 2 2];

% Re-order dimensions so that data array has dimensions [n_epochs, n_channels,
% n_times] as it's required by get_csp method
data = permute(data,[3 1 2]);

% Labels mean classes corresponding to each epoch
labels = [0 1];

% True values to compare with
filters_true = [-0.313406 0.0792153; -0.280803  -0.480046];
eigenvalues_true = [0.45671339 0.75297906];

% Calc filters and eigenvalues and compare them
[filters, eigenvalues] = DataFilter.get_csp(data, labels);

File: julia_package/brainflow/test/csp.jl
using BrainFlow

data = Array{Float64,3}(undef, 2, 4, 2)
data[:, :, 1] = [[6,3] [3,0] [1,5] [5,1]]
data[:, :, 2] = [[1,5] [5,1] [6,2] [2,2]]
data = permutedims(data, [3, 1, 2])
labels = [0.0, 1.0]

filters_true = [[-0.313406,-0.280803] [0.0792153, -0.480046]]
filters, eigvals = BrainFlow.get_csp(data, labels)

println("filters = ")
println(filters)

println("true filters =")
println(filters_true)